### PR TITLE
Use NPM templates for React

### PR DESCRIPTION
**Pull request by joebochill**
For PXBLUE-1723.

#### Changes proposed in this Pull Request:
- Add framework selector from Huayun's branch
- Integrate the React blank templates from NPM
- Reorganize the constants into separate files for readability

Published beta for testing at **@pxblue/cli@1.1.0-beta.0**

You can test the new templates by running:
```
// following on-screen prompts
npx -p @pxblue/cli@beta pxb new react
// or via flags
npx -p @pxblue/cli@beta pxb new react --language=ts --template=blank
// or
npx -p @pxblue/cli@beta pxb new react --language=js --template=blank
```

You can also leave off the template flag, as blank will be used by default

#### Screenshots
image

### CODE REVIEW

**Comment by huayunh** [comment]
Update readme

File: src/extensions/project-extensions.ts [hunk 1]
```diff
@@ -32,17 +40,28 @@ module.exports = (toolbox: GluegunToolbox): void => {
...
+        switch (template.toLocaleLowerCase()) {
+            case 'basic routing': // Coming Soon: templateName = isTs ? '@pxblue/routing-typescript' : '@pxblue/routing';
+            case 'routing': // to allow for --template=routing instead of --template="basic routing"
+            case 'authentication': // Coming Soon: templateName = isTs ? '@pxblue/authentication-typescript' : '@pxblue/authentication';
+            case 'blank':
+            default:
+                templateName = isTs ? '@pxblue/blank-typescript' : '@pxblue/blank';
+        }
+
+        const command = `npx create-react-app ${name} --template ${templateName}`;
```
Comment (huayunh) [inline src/extensions/project-extensions.ts:64]: We don't plan to allow them to just generate a regular plain react project?

Comment (huayunh) [reply src/extensions/project-extensions.ts:64]: Like a "no template" option?

Comment (huayunh) [reply src/extensions/project-extensions.ts:64]: Or does that complicate the logic too much?

**Review by huayunh — COMMENTED** [review]
All codes look good. Verified that the beta version works pretty smoothly.

File: src/extensions/project-extensions.ts (hunk 1 above)
Comment (joebochill) [reply src/extensions/project-extensions.ts:64]: I wasn't planning on it. If they want a plain react project they can just use `create-react-app project-name` and skip the PX Blue CLI altogether. 

There is really no advantage of the standard CRA template over our blank template (it's just a logo spinning in the middle of the screen with some accompanying styles).

Comment (joebochill) [reply src/extensions/project-extensions.ts:64]: With either the PXBlue Blank or CRA Blank, the first user step is "rip out the placeholder content and start building my actual app"

**Review by huayunh — APPROVED** [review]

**Comment by jeffvg** [comment]
looks good, pretty fast on both PC and Mac

**Pull request merged**